### ISSUE TITLE
Support unscaling grad on CPU

### ISSUE THREAD

**Issue by zhuzilin** [pytorch/pytorch#73833]
### 🚀 The feature, motivation and pitch

For now, the kernel for unscaling gradients `_amp_foreach_non_finite_check_and_unscale_` only has cuda version. However, as the scale of pretrained models are growing larger and larger, the solutions for reducing memory usage during training, e.g. [microsoft/DeepSpeed](https://github.com/microsoft/DeepSpeed), [Tencent/PatrickStar](https://github.com/Tencent/PatrickStar), [tunib-ai/oslo](https://github.com/tunib-ai/oslo), often offload the gradient to CPU and do optimizer updation there.

Therefore, I wonder if you have interest to add native support for unscaling gradient on CPU. In this way, the above frameworks could easily reuse the `autocast` mechanism and we could have coherent amp casting policy (whereas now, they are leaning to implement their own fp16 training).

I'd love to help if you are interested :)

### Additional context

The `_amp_foreach_non_finite_check_and_unscale_` is called here:
https://github.com/pytorch/pytorch/blob/c088c8fe92dae2a49d9c594d3af299ec05099210/torch/cuda/amp/grad_scaler.py#L201-L226

cc @VitalyFedyunin @mcarilli @ptrblck

**Comment by leslie-fang-intel** [comment]
It‘s a feature request with the supporting of FP16 training on CPU.

**Comment by CaoE** [comment]
Unscaling grad on CPU had been supported. https://github.com/pytorch/pytorch/issues/111559.
